lib/matplotlib/tests/test_axes.py

```diff
@@ -3696,6 +3696,28 @@ def test_step_linestyle():
         ax.set_ylim([-1, 7])
 
 
+@image_comparison(['step_histlike'], remove_text=True, extensions=['png'])
+def test_step_histlike():
```

Comment:
consider writing the test using check_figures_equal, which would better test that using step(..., where="between") is indeed equivalent to some call to plot()